tests/unit/test_git_hooks.py

```
@@ -12,10 +12,19 @@
     _generate_merged_branch_guard,
     _generate_pre_push_hook_script,
     _get_git_hooks_dir,
+    _global_hooks_dir,
     _parse_hook_info,
 )
 
 
+def _is_posix_sh(script: str) -> bool:
+    import tempfile
+
+    p = Path(tempfile.mktemp(suffix=".sh"))
```

Ruff (0.15.17)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)